src/core/agent/services/multiSigService.ts

```diff
@@ -744,6 +752,168 @@ class MultiSigService extends AgentService {
+    }
+    if (!ourIdentifier) {
+      throw new Error(MultiSigService.MEMBER_AID_NOT_FOUND);
+    }
+    return {
+      ourIdentifier,
+      multisigMembers,
+    };
+  }
+
+  async endRoleAuthorization(multisigSignifyName: string) {
+    const { ourIdentifier, multisigMembers } =
+      await this.getMultisigParticipants(multisigSignifyName);
+    const hab = await this.props.signifyClient
+      .identifiers()
+      .get(multisigSignifyName);
+    const aid = hab["prefix"];
+    const results = [];
+    const recp = multisigMembers
+      .filter((signing: any) => signing.aid !== ourIdentifier.id)
+      .map((member: any) => member.aid);
+    const ourAid: Aid = await this.props.signifyClient
+      .identifiers()
+      .get(ourIdentifier.signifyName as string);
+    for (const member of multisigMembers) {
+      const eid = Object.keys(member.ends.agent)[0]; //agent of member
+      const stamp = new Date().toISOString().replace("Z", "000+00:00");
+
+      const endRoleRes = await this.props.signifyClient
+        .identifiers()
+        .addEndRole(multisigSignifyName, "agent", eid, stamp);
+      const op = await endRoleRes.op();
+      const rpy = endRoleRes.serder;
+      const sigs = endRoleRes.sigs;
+      const mstate = hab["state"];
+      const seal = [
+        "SealEvent",
+        { i: hab["prefix"], s: mstate["ee"]["s"], d: mstate["ee"]["d"] },
+      ];
+      const sigers = sigs.map((sig) => new Siger({ qb64: sig }));
+      const roleims = d(
+        messagize(rpy, sigers, seal, undefined, undefined, false)
+      );
+      const atc = roleims.substring(rpy.size);
+      const roleEmbeds = {
+        rpy: [rpy, atc],
+      };
+
+      await this.sendMultisigExn(
+        ourIdentifier?.signifyName,
+        ourAid,
+        MultiSigRoute.RPY,
+        roleEmbeds,
+        recp,
+        { gid: aid }
+      );
+      results.push(op);
+    }
+    return results;
+  }
+
+  async joinAuthorization(notificationSaid: string) {
+    const exchangeMessage = await this.props.signifyClient
+      .exchanges()
+      .get(notificationSaid);
+    if (!exchangeMessage) {
+      throw new Error(
+        `${MultiSigService.EXN_MESSAGE_NOT_FOUND} ${notificationSaid}`
+      );
+    }
+    const multisigAid = exchangeMessage.exn.a.gid;
+    const multisigMetadataRecord =
+      await this.identifierStorage.getIdentifierMetadata(multisigAid);
+    const multisigSignifyName = multisigMetadataRecord.signifyName;
+    const request = await this.props.signifyClient
+      .groups()
+      .getRequest(notificationSaid)
+      .catch((error) => {
+        const errorStack = (error as Error).stack as string;
+        const status = errorStack.split("-")[1];
+        if (/404/gi.test(status) && /SignifyClient/gi.test(errorStack)) {
+          return [];
+        } else {
+          throw error;
+        }
+      });
+    if (!request.length) {
+      throw new Error(
+        `${MultiSigService.GROUP_REQUEST_NOT_FOUND} ${notificationSaid}`
```

Comment:
I think we can just remove the try catch on `getRequest`. The reason is we expect that it will exist on KERIA - and we will get an error anyway from Signify.

The scenario where we want to catch and return [] or undefined is if we aren't sure, and we will continue on doing something instead of throwing an error straight away (like this `GROUP_REQUEST_NOT_FOUND`)